### PR TITLE
Shared model/new yac hash

### PR DESCRIPTION
**Pull request by x3medima17**
### Description of the Change
Moves `YacHash` and `YacHashProvider` to the new model.
Due to the fact that I split moving consensus to the new model in 4 Pull Requests, `VoteMessage` will be migrated in the last PR. So this check will be removed in the last one.
https://github.com/hyperledger/iroha/blob/40a237038db5b94bf5e54b706562e7a6b7c3d1c6/irohad/consensus/yac/transport/yac_pb_converters.hpp#L42
Depends on #1052 

### Benefits
Less dependency on old model

### Possible Drawbacks 
None

### CODE REVIEW

File: irohad/consensus/yac/impl/yac_hash_provider_impl.cpp [hunk 1]
```diff
@@ -17,26 +17,30 @@
...
 
 namespace iroha {
   namespace consensus {
     namespace yac {
 
-      YacHash YacHashProviderImpl::makeHash(const model::Block &block) const {
+      YacHash YacHashProviderImpl::makeHash(
+          const shared_model::interface::Block &block) const {
         YacHash result;
         // TODO 01/08/17 Muratov: add proposal hash to block,
```
Comment (kamilsa) [inline irohad/consensus/yac/impl/yac_hash_provider_impl.cpp]: This IR is closed

File: irohad/consensus/yac/transport/impl/network_impl.hpp [hunk 2]
```diff
@@ -91,12 +88,13 @@ namespace iroha {
          * peers map
          * @param peer to instantiate connection with
          */
-        void createPeerConnection(const model::Peer &peer);
+        void createPeerConnection(const shared_model::interface::Peer &peer);
 
         /**
          * Mapping of peer objects to connections
```
Comment (kamilsa) [inline irohad/consensus/yac/transport/impl/network_impl.hpp]: Fix comment

File: irohad/consensus/yac/transport/yac_pb_converters.hpp [hunk 3]
```diff
@@ -35,10 +37,22 @@ namespace iroha {
...
 
           auto block_signature = hash->mutable_block_signature();
-          block_signature->set_signature(
-              vote.hash.block_signature.signature.to_string());
-          block_signature->set_pubkey(
-              vote.hash.block_signature.pubkey.to_string());
+
+          //Will fix it in the next PR, very soon, don't worry
+          if (vote.hash.block_signature == nullptr) {
+            auto sig = shared_model::proto::SignatureBuilder()
```
Comment (kamilsa) [inline irohad/consensus/yac/transport/yac_pb_converters.hpp]: There is a `builders/common_objects/signature_builder.hpp` which returns interface signature. Please add this builder to `default_builders.hpp` as it is done for other builders.

File: irohad/consensus/yac/transport/yac_pb_converters.hpp [hunk 4]
```diff
@@ -52,12 +66,16 @@ namespace iroha {
           VoteMessage vote;
           vote.hash.proposal_hash = pb_vote.hash().proposal();
           vote.hash.block_hash = pb_vote.hash().block();
-          vote.hash.block_signature.signature =
-              *stringToBlob<iroha::sig_t::size()>(
-                  pb_vote.hash().block_signature().signature());
-          vote.hash.block_signature.pubkey =
-              *stringToBlob<iroha::pubkey_t::size()>(
-                  pb_vote.hash().block_signature().pubkey());
+          auto sig = shared_model::proto::SignatureBuilder()
```
Comment (kamilsa) [inline irohad/consensus/yac/transport/yac_pb_converters.hpp]: Please use interface builder as well

File: irohad/consensus/yac/yac.hpp [hunk 5]
```diff
@@ -106,17 +102,21 @@ namespace iroha {
          * top block in ledger does not correspond to consensus round number
          */
 
-        void applyCommit(boost::optional<model::Peer> from,
+        void applyCommit(boost::optional<std::shared_ptr<shared_model::interface::Peer>> from,
```
Comment (kamilsa) [inline irohad/consensus/yac/yac.hpp]: Not clang-formatted

File: test/module/irohad/consensus/yac/peer_orderer_test.cpp [hunk 6]
```diff
@@ -58,9 +54,13 @@ class YacPeerOrdererTest : public ::testing::Test {
   std::vector<std::shared_ptr<shared_model::interface::Peer>> peers = [] {
     std::vector<std::shared_ptr<shared_model::interface::Peer>> result;
     for (size_t i = 1; i <= N_PEERS; ++i) {
-      auto peer = std::shared_ptr<shared_model::interface::Peer>(shared_model::proto::PeerBuilder()
-                      .address(std::to_string(i)).pubkey(shared_model::interface::types::PubkeyType(std::string(32, '0')))
-                      .build().copy());
+      auto peer = std::shared_ptr<shared_model::interface::Peer>(
```
Comment (kamilsa) [inline test/module/irohad/consensus/yac/peer_orderer_test.cpp]: Use `DefaultPeerBuilder` instead

File: test/module/irohad/consensus/yac/yac_mocks.hpp [hunk 7]
```diff
@@ -35,12 +37,16 @@
...
-        model::Peer peer;
-        peer.address = address;
-        // TODO: 19.01.2019 kamil substitute with function, IR-813
-        std::copy(address.begin(), address.end(), peer.pubkey.begin());
-        return peer;
+      std::shared_ptr<shared_model::interface::Peer> mk_peer(
+          const std::string &address) {
+        auto key = std::string(32, '0');
+        std::copy(address.begin(), address.end(), key.begin());
+        auto ptr = shared_model::proto::PeerBuilder()
```
Comment (kamilsa) [inline test/module/irohad/consensus/yac/yac_mocks.hpp]: Use DefaultPeerBuilder

File: irohad/consensus/yac/impl/cluster_order.cpp [hunk 8]
```diff
@@ -34,8 +34,8 @@ namespace iroha {
           std::vector<std::shared_ptr<shared_model::interface::Peer>> order)
           : order_(std::move(order)) {}
 
-      // TODO :  24/03/2018 x3medima17: make it const, IR-1164
-      const shared_model::interface::Peer &ClusterOrdering::currentLeader() {
+    // TODO :  24/03/2018 x3medima17: make it const, IR-1164
```
Comment (nickaleks) [inline irohad/consensus/yac/impl/cluster_order.cpp]: Remove todo?

File: irohad/consensus/yac/transport/impl/network_impl.hpp [hunk 9]
```diff
@@ -45,12 +45,9 @@ namespace iroha {
...
             std::shared_ptr<YacNetworkNotifications> handler) override;
-        void send_commit(const shared_model::interface::Peer &to,
-                         const CommitMessage &commit) override;
-        void send_reject(const shared_model::interface::Peer &to,
-                         RejectMessage reject) override;
-        void send_vote(const shared_model::interface::Peer &to,
-                       VoteMessage vote) override;
+        void send_commit(const shared_model::interface::Peer &to,const CommitMessage &commit) override;
+        void send_reject(const shared_model::interface::Peer &to, RejectMessage reject) override;
+        void send_vote(const shared_model::interface::Peer &to, VoteMessage vote) override;
```
Comment (nickaleks) [inline irohad/consensus/yac/transport/impl/network_impl.hpp]: Clang format aswell

File: irohad/consensus/yac/transport/yac_network_interface.hpp [hunk 10]
```diff
@@ -67,8 +67,7 @@ namespace iroha {
          * @param to - peer recipient
          * @param commit - message for sending
          */
-        virtual void send_commit(const shared_model::interface::Peer &to,
-                                 const CommitMessage &commit) = 0;
+        virtual void send_commit(const shared_model::interface::Peer &to,const CommitMessage &commit) = 0;
```
Comment (nickaleks) [inline irohad/consensus/yac/transport/yac_network_interface.hpp]: Clang format pls

File: irohad/consensus/yac/yac.hpp [hunk 11]
```diff
@@ -102,17 +102,11 @@ namespace iroha {
...
-                from,
-            const CommitMessage &commit);
-        void applyReject(
-            boost::optional<std::shared_ptr<shared_model::interface::Peer>>
-                from,
-            const RejectMessage &reject);
-        void applyVote(boost::optional<
-                           std::shared_ptr<shared_model::interface::Peer>> from,
-                       const VoteMessage &vote);
+        void applyCommit(boost::optional<std::shared_ptr<shared_model::interface::Peer>> from,
```
Comment (nickaleks) [inline irohad/consensus/yac/yac.hpp]: Clang format pls

File: irohad/consensus/yac/impl/yac_gate_impl.cpp [hunk 12]
```diff
@@ -30,6 +30,12 @@
 #include "simulator/block_creator.hpp"
 #include "utils/polymorphic_wrapper.hpp"
```
Comment (nickaleks) [inline irohad/consensus/yac/impl/yac_gate_impl.cpp]: Why whitespace between includes here?

File: irohad/consensus/yac/impl/yac_hash_provider_impl.cpp [hunk 13]
```diff
@@ -17,26 +17,28 @@
 
 #include "consensus/yac/impl/yac_hash_provider_impl.hpp"
```
Comment (nickaleks) [inline irohad/consensus/yac/impl/yac_hash_provider_impl.cpp]: Please separate main include from the rest

**Pull request merged**